src/modules/esl-utils/async/test/debounce.test.ts

```diff
@@ -40,12 +40,12 @@ describe('async/debounce', () => {
   test('cancel debounce', () => {
     const fn = jest.fn();
     const debounced = debounce(fn, 10);
-    const promise = debounced.promise;
     debounced();
-    debounced.cancel();
-
+    setTimeout(() => debounced.cancel());
+    const promise = debounced.promise;
```

Comment:
That's incorrect: 
`debounced.cancel()` should be called sync just after the first call
In case we call it in bounds of a separate task - that's a different type of test case.

```suggestion
    const promise = debounced.promise;
    debounced.cancel();
```